core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -593,7 +593,7 @@ class KafkaApis(val requestChannel: RequestChannel,
 
     val unauthorizedTopicMetaData = unauthorizedTopics.map(topic => new TopicMetadata(topic, Seq.empty[PartitionMetadata], ErrorMapping.AuthorizationCode))
 
-    val topicMetadata = getTopicMetadata(authorizedTopics, request.securityProtocol)
+    val topicMetadata = if (authorizedTopics.isEmpty) Seq.empty[TopicMetadata] else getTopicMetadata(authorizedTopics, request.securityProtocol)
```

Comment:
+1,  Jun pointed this issue out to me in the initial PR and I thought I handled it correctly, Thank god for test cases but I feel like a complete idiot for missing this even after being specifically told :-(.